x/leverage/keeper/msg_server.go

```diff
@@ -114,7 +121,8 @@ func (s msgServer) MaxWithdraw(
 	)
 
 	if uToken.IsZero() {
-		return nil, types.ErrMaxWithdrawZero
+		zeroCoin := sdkutil.ZeroCoin(msg.Denom)
+		return &types.MsgMaxWithdrawResponse{Withdrawn: zeroCoin, Received: zeroCoin}, nil
```

Comment:
`Withdrawn` is a uToken field and `Received` is a token field - they cannot have the same denom

Reply:
Good point, will update. So we need to add more tests ;)

Reply:
Ooo, we had a test, but it was buggy!